pkg/scheduler/framework/types.go

```diff
@@ -350,6 +350,9 @@ type Diagnosis struct {
 	PreFilterMsg string
 	// PostFilterMsg records the messages returned from PostFilter plugins.
 	PostFilterMsg string
+	// AbsentNodesStatus defines a status for all UnschedulableAndUnresolvable nodes
+	// that are absent in NodeToStatusMap.
+	AbsentNodesStatus *Status
```

Comment:
What do you think about coupling these together (AbsentNodesStatus and NodeToStatusMap) - perhaps by changing NodeToStatusMap to a struct?  `set`, `setAbsentStatus`, and `get` (which returns the absent status, if the value is not explicitly defined) would make it easier to use.

It does change the interface, but on the other hand this is going out in a new release, would give plugin developers a heads up to the semantics change, and provide us with future flexibility that the map type doesnt give.

Reply:
That would be good and increase readability. The question is, if we can change the plugins interface that deep. @alculquicondor what do you think?

Reply:
Created second PR #126022 to discover the possible solution you proposed here.